File: js-port/funcscript-js/src/funcs/text/upper-text-function.js
const { BaseFunction, CallType } = require('../../core/function-base');
const helpers = require('../helpers');
const { FSDataType } = require('../../core/fstypes');

class UpperTextFunction extends BaseFunction {
  constructor() {
    super();
    this.symbol = 'upper';
    this.callType = CallType.Prefix;
  }

  get maxParameters() {
    return 1;
  }

  evaluate(provider, parameters) {
    const error = helpers.expectParamCount(this.symbol, parameters, this.maxParameters);
    if (error) {
      return error;
    }

    const value = helpers.ensureTyped(parameters.getParameter(provider, 0));
    if (helpers.typeOf(value) === FSDataType.Null) {
      return helpers.typedNull();
    }
    if (helpers.typeOf(value) !== FSDataType.String) {
      return helpers.makeError(helpers.FsError.ERROR_TYPE_MISMATCH, `${this.symbol}: string parameter expected`);
    }

    const text = helpers.valueOf(value);
    return helpers.makeValue(FSDataType.String, text.toUpperCase());
  }

  parName(index) {
    return index === 0 ? 'text' : '';
  }
}

module.exports = {
  UpperTextFunction
};

File: FuncScript/Functions/Text/UpperTextFunction.cs
using FuncScript.Core;
using FuncScript.Error;
using FuncScript.Model;

namespace FuncScript.Functions.Text
{
    [ProviderCollection("text")]
    public class UpperTextFunction : IFsFunction
    {
        public int MaxParsCount => 1;

        public CallType CallType => CallType.Prefix;

        public string Symbol => "upper";

        public int Precedence => 0;

        public object Evaluate(object par)
        {
            var pars = FunctionArgumentHelper.ExpectList(par, this.Symbol);

                        if (pars.Length != 1)
                throw new Error.TypeMismatchError($"{this.Symbol}: single string parameter expected");

            var value = pars[0];

            if (value == null)
                return null;

            if (value is not string text)
                throw new Error.TypeMismatchError($"{this.Symbol}: string parameter expected");

            return text.ToUpperInvariant();
        }

        public string ParName(int index)
        {
            return index == 0 ? "text" : string.Empty;
        }
    }
}
